pkg/infra/iac2/templates_compiler.go

```diff
@@ -147,6 +147,11 @@ func (tc templatesCompiler) renderResource(out io.Writer, resource core.Resource
 	inputArgs := make(map[string]string)
 	var zeroValue reflect.Value
 	for fieldName := range tmpl.inputTypes {
+		// dependsOn will be a reserved field for us to use to map dependencies. If specified as an Arg we will automatically call resolveDependencies
+		if fieldName == "dependsOn" {
```

Comment:
I'd moderately prefer if we did it based on type, not fieldName — just feels a bit easier to track through the code. That is, create in go a `type Dependencies struct{}`, and then check here:

```go
if _, ok := childVal.Interface().(core.Dependencies); ok {
```

Not a blocker.

Reply:
but we wont have that struct used within the actual IR structs so we would never match to it. Its odd to have a struct created and appended to every single IR module if we need dependencies. These are also represented in the graph which makes it weird. Open to another way of doing this though, but it seems to me like the logic should reside completely in the IaC plugin since we already map dependencies in the graph

Reply:
Oh, yeah good point! Alright, ignore me. :)